backend/controller/door.go

```diff
@@ -0,0 +1,38 @@
+package controller
+
+import (
+	"api/helpers"
+	"api/models"
+	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+func GetAllDoors(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	err, doors := helpers.GetAllTable(w, []models.Door{})
+	if err == nil {
+		helpers.JsonWriter(w, doors)
+	}
+}
+
+func CreateDoor(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var role models.Door
+	err := json.NewDecoder(r.Body).Decode(&role)
```

Comment:
Why would you define err if its unused? Also if the client sends incorrectly formatted payload we should tell them.

Reply:
Its used read the create new record function, I could probably rework the logic but you'll notice I passed it in, and thats cuz I wanted the create new Records function to be able to handle cases where you got the req body other ways and hence removing that logic from it, but this error is handled if you look in there

Reply:
I see